core/sql/common/MiscType.cpp

```diff
@@ -106,21 +172,51 @@ const NAType* SQLBoolean::synthesizeType(enum NATypeSynthRuleEnum synthRule,
   // expressions.
   //
   if (synthRule == SYNTH_RULE_UNION)
-    return new(h) SQLBoolean();
+    return new(h) SQLBooleanNative
+      (operand1.supportsSQLnull() || operand2.supportsSQLnull());
+
   return NULL;
 } // synthesizeType()
 
+void SQLBooleanNative::minRepresentableValue(void* bufPtr, Lng32* bufLen,
+                                     NAString ** stringLiteral,
+				     CollHeap* h) const
+{
+  assert(*bufLen >= sizeof(Int8));
+  Lng32 valueBuf;
```

Comment:
It doesn't hurt anything, but it looks like valueBuf is not used other than setting it to zero three lines down.